src/main/application-update/install-application-update-tray-item.injectable.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { trayMenuItemInjectionToken } from "../tray/tray-menu-item/tray-menu-item-injection-token";
+import quitAndInstallUpdateInjectable from "../electron-app/features/quit-and-install-update.injectable";
+import discoveredUpdateVersionInjectable from "../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import updateIsBeingDownloadedInjectable from "../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import { withErrorSuppression } from "../../common/utils/with-error-suppression/with-error-suppression";
+import { pipeline } from "@ogre-tools/fp";
+import withErrorLoggingInjectable from "../../common/utils/with-error-logging/with-error-logging.injectable";
+
+const installApplicationUpdateTrayItemInjectable = getInjectable({
+  id: "install-update-tray-item",
+
+  instantiate: (di) => {
+    const quitAndInstallUpdate = di.inject(quitAndInstallUpdateInjectable);
+    const discoveredVersionState = di.inject(discoveredUpdateVersionInjectable);
+    const downloadingUpdateState = di.inject(updateIsBeingDownloadedInjectable);
+    const withErrorLoggingFor = di.inject(withErrorLoggingInjectable);
+
+    return {
+      id: "install-update",
+      parentId: null,
+      orderNumber: 50,
+
+      label: computed(() => {
+        const versionToBeInstalled = discoveredVersionState.value.get()?.version;
+
+        return `Install update ${versionToBeInstalled}`;
+      }),
+
+      enabled: computed(() => true),
+
+      visible: computed(
+        () => !!discoveredVersionState.value.get() && !downloadingUpdateState.value.get(),
+      ),
```

Comment:
Given that `visible` is meaningless if `enabled` returns `false`. Would a better design (which only covers the three reasonable cases) be something like the following (in conjunction with the above label):

```ts
export type DisplayConfiguration = {
  visibility: "hidden" | "separator";
} | {
  visibility: "disabled" | "enabled";
  label: string;
};
```

This way we only need to care about the shape of the label when it matters, and we don't get the strange checks like we do on line 39.

The above could then be written nice an succinctly as the following:

```ts
return {
  id: "install-update",
  displayConfiguration: computed(() => {
    const discoveredVersion = discoveredVersionState.value.get();
    const downloadingUpdate = downloadingUpdateState.value.get();

    if (discoveredVersion === null || !downloadingUpdate) {
      return { visibility: "hidden" };
    }

    return {
      visibility: "enabled",
      label: `Install update ${discoveredVersion.version}`,
    };
  }),
};
```

Reply:
Typing does not make it less magical string. Usages of magical strings lead to non OCP code and should be avoided wherever possible. 

e.g. typing might be "good", but still, code isn't.

```typescript
if (visibility === 'hidden') {
  // do something
}
```

Reply:
Not everything has to be OCP. Especially here where there are no object hierarchies at all.

Fine, if you don't like strings we can have the following enum instead

```ts
export enum MenuItemDisplay {
  HIDDEN,
  SEPARATOR,
  ENABLED,
  DISABLED,
}
```

Reply:
@Iku-turso Please don't just resolve conversations (unless you are using resolution as a means of marking a TODO item as complete before pushing of course).

Reply:
Hmm. That's weird, I remember writing a legitimate reason for us resolving this. Oh well.

Reply:
Resolved this conversation privately by making sure that everyone understands that we are not looking for solutions that are written in stone. We are looking for solutions that can later be improved easily when the requirement comes. 

That being said, for now, it's good enough since behaviour is green. Future requirements will re-shape the implementation if needed.